File: services/workflow-automation-service/src/models/approval.rs
use serde::Deserialize;
use serde_json::Value;
use uuid::Uuid;

#[derive(Debug, Deserialize)]
pub struct InternalApprovalContinuationRequest {
    pub workflow_id: Uuid,
    pub workflow_run_id: Uuid,
    pub step_id: String,
    pub decision: String,
    #[serde(default)]
    pub context: Value,
}

File: services/workflow-automation-service/internal/models/approval.go
package models

import (
	"encoding/json"

	"github.com/google/uuid"
)

// InternalApprovalContinuationRequest mirrors
// models::approval::InternalApprovalContinuationRequest. The payload of
// `POST /api/v1/workflows/approvals/{id}/continue`.
type InternalApprovalContinuationRequest struct {
	WorkflowID    uuid.UUID       `json:"workflow_id"`
	WorkflowRunID uuid.UUID       `json:"workflow_run_id"`
	StepID        string          `json:"step_id"`
	Decision      string          `json:"decision"`
	Context       json.RawMessage `json:"context,omitempty"`
}
